test/fgpu/test_postproc.py

```diff
@@ -113,6 +119,18 @@ def postproc_host(
     return np.stack(out, axis=3), np.stack(saturated, axis=1)
 
 
+def unpack_complex(data: np.ndarray) -> np.ndarray:
+    """Unpack array of Gaussian integers to complex dtype."""
+    if data.dtype.itemsize == 1:
+        # It's 4-bit packed. We assume that >> will sign extend on signed types
+        real = data.view(np.int8) >> 4
+        imag = data.view(np.int8) << 4 >> 4
```

Comment:
Hang on, what is this sorcery? I gather it has to do with 
> We assume that >> will sign extend on signed types

Pasting the PR blurb for ease of reference, although I can't quite figure out how that influences the one-liner `<< 4 >> 4`
> It's implemented with the real part in the high bits and imaginary in the low bits of each byte.
That's consistent with SPEAD's interpretation of arrays with sub-byte types (big-endian)

Reply:
Let's say that the value is 0bABCDEFGH i.e. the real part is 0bABCD and the imaginary part is 0bEFGH. Then `x << 4` is 0bEFGH0000 (the top 4 bits overflow and are discarded) and then x >> 4 is 0bEEEEEFGH, which is the imaginary part sign-extended to 8 bits.

Reply:
*Interesting*, I (clearly) was not aware of the concept of "sign extension", appreciate the explanation.